sda-download/api/sda/sda_test.go

```diff
@@ -552,3 +566,217 @@ func TestEncrypted_Coords(t *testing.T) {
 	_, _, err = calculateEncryptedCoords(0, 0, "bytes=start-end", fileDetails)
 	assert.Error(t, err)
 }
+
+type fakeGRPC struct {
+	t      *testing.T
+	pubkey [32]byte
+}
+
+func (f *fakeGRPC) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	// Poor person's grpc server
+
+	w.Header().Add("Content-type", "application/grpc+proto")
+	w.Header().Add("Grpc-encoding", "identity")
+	w.Header().Add("Trailer", "Grpc-status")
+
+	var inLength int32
+	compressed := make([]byte, 1)
+	_, err := r.Body.Read(compressed)
+	assert.NoError(f.t, err, "Could not read compressed flag")
+	assert.Equal(f.t, compressed[0], byte(0), "Unexpected compressed flag")
+
+	err = binary.Read(r.Body, binary.BigEndian, &inLength)
+	assert.NoError(f.t, err, "Could not read length")
+
+	body, err := io.ReadAll(r.Body)
+	assert.NoError(f.t, err, "Could not read body")
+
+	re := reencrypt.ReencryptRequest{}
+	err = proto.Unmarshal(body, &re)
+	assert.NoError(f.t, err, "Could not unmarshal request")
+
+	rr := reencrypt.ReencryptResponse{Header: re.GetOldheader()}
+	response, err := proto.Marshal(&rr)
+	assert.NoError(f.t, err, "Could not marshal response")
+
+	w.WriteHeader(200)
+	_, err = w.Write([]byte{0})
+	assert.NoError(f.t, err, "Could not write response flag")
+
+	err = binary.Write(w, binary.BigEndian, int32(len(response)))
+	assert.NoError(f.t, err, "Could not write response length")
+
+	_, err = w.Write(response)
+	assert.NoError(f.t, err, "Could not write response")
+	w.Header().Add("Grpc-status", "0")
+}
+
+func TestDownload_Success_Whole(t *testing.T) {
```

Comment:
It seems this test function contains many scenarios. Not sure if it is better to divide them into separate functions, but if we keep them in one function, maybe change the function name to e.g. `Test_Download_Whole_Or_Range` to better reflect the testing scenarios?

Reply:
Yes, it turned out to be a bit more setup than I intended from the start so it felt like it made sense to test more cases. I guess one could factor most of that out but there'll still be a fair bit of setup so I think it's best to keep it as is for now, updated the name.